src/ASM/BlockElmMats.C

```diff
@@ -168,12 +171,16 @@ const Matrix& BlockElmMats::getNewtonMatrix () const
   std::cout <<"Resulting Newton matrix"<< A.front();
 #endif
 
+  newtonMatAssembled = true;
```

Comment:
So where are the bool members reset to false again when assembling a new system? Ideally, that should be handled my a member method as well.

Reply:
it's not really necessary as we never reuse LocaIntegrals. I can add it of course, but it will be completely unused.

Reply:
maybe overriding redim is sufficient?

Reply:
I see,its fine as is then, although I don't see the big picture why this is needed since the LocaIntegral objects are deleted after each element in any case.

Reply:
when we assemble the rhs vector using the element matrices, getNewtonMatrix() is called, https://github.com/OPM/IFEM/blob/master/src/ASM/AlgEqSystem.C#L171

which would overwrite the cached matrix set in initElement (unless we instead cache all the blocks which is not very convenient).